src/Servers/Kestrel/Core/src/Internal/Http/Http1Connection.cs

```diff
@@ -667,6 +670,14 @@ protected override bool TryParseRequest(ReadResult result, out bool endConnectio
             }
             throw;
         }
+#pragma warning disable CS0618 // Type or member is obsolete
+        catch (BadHttpRequestException ex)
```

Comment:
The error in this scenario is already `Unrecognized HTTP version: 'HTTP/2.0'`, the detection is half done.
```suggestion
        catch (BadHttpRequestException ex) when (ex.Reason == RequestRejectionReason.UnrecognizedHTTPVersion)
```
You could also probably match directly on the exception message rather than re-checking the preface.

Reply:
I considered checking the error message, but the exception message only includes error detail if _showDetailedError is true, which is only true when log level is Info or higher.